src/java.desktop/share/native/common/java2d/opengl/OGLBlitLoops.c

```diff
@@ -406,22 +409,24 @@ OGLBlitSwToTexture(SurfaceDataRasInfo *srcInfo, OGLPixelFormat *pf,
 
     // in case pixel stride is not a multiple of scanline stride the copy
     // has to be done line by line (see 6207877)
-    if (srcInfo->scanStride % srcInfo->pixelStride != 0) {
-        jint width = dx2 - dx1;
-        jint height = dy2 - dy1;
-        GLvoid *pSrc = srcInfo->rasBase;
-
-        while (height > 0) {
-            j2d_glTexSubImage2D(dstOps->textureTarget, 0,
+    if (pf != NULL) {
```

Comment:
https://github.com/openjdk/jdk/blob/c96d09acd95d0ccf2fef50b8ccfeb5e2a0aa0968/src/java.desktop/share/native/common/java2d/opengl/OGLBlitLoops.c#L651

OGLBlitSwToTexture is called with PixelFormat defined as

https://github.com/openjdk/jdk/blob/c96d09acd95d0ccf2fef50b8ccfeb5e2a0aa0968/src/java.desktop/share/native/common/java2d/opengl/OGLSurfaceData.c#L52-L76

which is not null so not sure why pf will ever be null? Did you see it null anytime?

Reply:
This was discussed already. There's another call site that passes NULL.

Reply:
Oh this is a different function. Not the one previously discussed, looks like it was added to the PR later.
This case at least *currently* cannot be called with NULL for the reason above.